android/app/src/main/java/app/covidshield/module/CovidShieldModule.kt

```diff
@@ -34,48 +34,30 @@ class CovidShieldModule(context: ReactApplicationContext) : ReactContextBaseJava
     }
 
     @ReactMethod
-    fun downloadDiagnosisKeysFiles(url: String, promise: Promise) {
+    fun downloadDiagnosisKeysFile(url: String, promise: Promise) {
         launch(Dispatchers.IO) {
-            var bufferedSource: BufferedSource? = null
             try {
                 val request = Request.Builder().url(url).build()
                 val response = okHttpClient.newCall(request).execute().takeIf { it.code() == 200 }
                     ?: throw Error("Network error")
-                bufferedSource = response.body()?.source() ?: throw Error("Network error")
-                val files = mutableListOf<ByteArray>()
-                while (true) {
-                    val size = bufferedSource.read(4)?.let { BigInteger(it).toInt() } ?: break
-                    val file = bufferedSource.read(size) ?: break
-                    files.add(file)
-                }
-                val fileDirs = writeFiles(files)
-                promise.resolve(fileDirs.toWritableArray())
+                val bytes = response.body()?.bytes() ?: throw Error("Network error")
+
+                val fileName = writeFile(bytes)
+                promise.resolve(fileName)
             } catch (exception: Exception) {
                 promise.reject(exception)
-            } finally {
-                bufferedSource?.close()
             }
         }
     }
```

Comment:
@henrytao-me do these changes make sense to you?

Reply:
How's big is the response payload? `response.body()?.bytes()` This will download all at once.

Reply:
The limit on Android is 2GB.

Reply:
the are definitely less than that.